x-pack/legacy/plugins/uptime/public/components/functional/monitor_list/to_condensed_check.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { Check } from '../../../../common/graphql/types';
+import { CondensedCheck } from './types';
+
+const inferCondensedFields = (
+  check: CondensedCheck,
+  currentStatus: string,
+  currentTimestamp: string
+) => {
+  const { status: condensedStatus, timestamp } = check;
+  if (condensedStatus !== currentStatus && condensedStatus !== 'mixed') {
```

Comment:
This logic is complex, could use a comment explaining it. I'm not immediately sure what this condition is trying to do.